tests/test_run_engine.py

```diff
@@ -139,3 +142,30 @@ def test_during_task_does_wait_with_small_timeout():
 
     event.wait.assert_called_with(0.1)
     assert event.wait.call_count == 2
+
+
+def test_run_engine_logs_all_documents(RE):
```

Comment:
I wonder whether this test would be better in a new `test_document_logging_callback.py` file?

The usual idea is to have a roughly 1-1 correspondence between implementation files and test files.

(we can chat about this if you want - it'll involve moving the `RE` fixture into a `conftest.py`)